### ISSUE TITLE
New posts are in random order

### ISSUE THREAD

**Issue by gdt** [pixelfed/pixelfed#4619]
Using pixelfed.social today, and uploading with firefox.   I selected mutiple pictures and made a post.  The filenames have camera serial numbers, the exif DateTimeOriginal tags are right (in UTC) , and the files have unix mtimes matching exif.    So regardless of whether it's by filename, or by date, I expect them in chronological order.   For some, I just ignore it, because it doesn't matter, but for a number of them, documenting an event, that's not ok, and I have to go manually sort.

I think the posts should be in either filename order, in exif DateTimeOriginal order, or perhaps in mtime order, by default.

I realize there is some concept that it follows the file picker order, but in my case the first file was highlighted and I shift clicked on the last, selecting them all.   That should lead to them being in order, somehow.

I found #3708 as fixed, so I wonder if this is a regression.

**Comment by gdt** [comment]
Really?  This is a remarkably user-unfriendly experience as it is.

I interpret a summary closed of "not planned" as "even if we got a PR we wouldn't merge it, so don't bother".

**Comment by trwnh** [comment]
GitHub puts a lot of things under the umbrella of "not planned", including duplicates, invalid, inactionable, etc etc. If a PR was made that targeted a demonstrable issue, it would of course be evaluated like any other PR.

**Comment by gdt** [comment]
Thanks.  What then is the closure reason?  I realize maintainers are busy (I maintain other software), but as a potential contributor I feel pretty discouraged.

Do the maintainers agree that the current behavior is suboptimal and that time order would be better?

**Comment by trwnh** [comment]
There's not anything that can be done on the upload file picker side I think, because that HTTP request usually happens in the order you select the photos and not in any sorted order. If there's an issue with selection order not being respected (which it was in my brief testing), I can reopen this issue. But otherwise, it could maybe be addressed by a feature request to have automatic sort options *after* upload.

**Comment by gdt** [comment]
Agreed it's hard to figure out how things are working now.  I just wanted to make sure this was "we don't think we're going to get to this" vs "we don't agree with what you want to do".    I'll do some selection order testing and post back with results - it makes sense to call that a defect and "sort by date" as an enhancement.

**Comment by gdt** [comment]
With 8 files that sort the same way in filename, exif date, and unix date, I created a new post.  I clicked the first one, and then shift-clicked the last one.    The images were in the post in an order that I can't explain at all.   Browser was firefox 128.0.3 on NetBSD.   So that's a data point that the file picker order was not respected.    However, it was shift-click, not command-click in order.

(I'll do another test with just three pictures that are obviously 1, 2 and 3 from content, and I'll leave it misordered and link to it.)
